### ISSUE TITLE
Work with lifetimes

### ISSUE THREAD

**Issue by aleksmelnikov** [Eolu/enum-assoc#2]
Hi. 
There're three error types in my example. Error and Error2 work, Error3 doesn't. How could I work with Error3? Does the crate support it? 
Thank you.

```
use thiserror::Error;
use enum_assoc::Assoc;

#[derive(Error, Debug, Assoc)]
pub enum Error {
    #[error("Unable to delete {:?} file.", .file)]
    ErrFileDelete { file: String },
}

#[derive(Error, Debug)]
pub enum Error2<'a> {
    #[error("Unable to delete {:?} file.", .file)]
    ErrFileDelete { file: &'a str },
}

#[derive(Error, Debug, Assoc)]
pub enum Error3<'a> {
    #[error("Unable to delete {:?} file.", .file)]
    ErrFileDelete { file: &'a str },
}

fn main(){
    let error = Error::ErrFileDelete{ file: "myfile".to_string() };
    println!("{}", error);

    let error = Error2::ErrFileDelete{ file: "myfile" };
    println!("{}", error);

    let error = Error3::ErrFileDelete{ file: "myfile" };
    println!("{}", error);
}
```
When compiling I get:
```
error[E0726]: implicit elided lifetime not allowed here
  --> test_enum_assoc_1/src/main.rs:18:10
   |
18 | pub enum Error3<'a> {
   |          ^^^^^^- help: indicate the anonymous lifetime: `<'_>`
   |
   = note: assuming a `'static` lifetime...

error: could not compile `test_enum_assoc_1` due to previous error
```

**Comment by Eolu** [comment]
Fixed in 0.3.4.

**Comment by aleksmelnikov** [comment]
It works now. Thanks a lot.
